### ISSUE TITLE
Change plot colour

### ISSUE THREAD

**Issue by xchoo** [nengo/nengo_gui#400]
It's really nifty that we can overlay plots, but having all the plots be the same colour makes it confusing. Being able to change the colour of individual plots (I guess they can only be 1D plots??) would be nice.

**Comment by tbekolay** [comment]
As an alternative, if we want to make colors distinguishable, we could have a single color index counter for value plot colors, rather than a counter for each individual value plot.

**Comment by xchoo** [comment]
Yeah! That works! :smile_cat:

**Comment by jgosmann** [comment]
But we only have like 6 colors, so they will repeat and might match up in exactly those plots you want to overlay.

**Comment by tcstewar** [comment]
> As an alternative, if we want to make colors distinguishable, we could have a single color index counter for value plot colors, rather than a counter for each individual value plot.

Hmm...  we'd need to also add a legend (on a mouseover popup?) so that we know the order for that graph.  How it is now, we at least can know that the order is blue, green, red, etc...

That said, I do like the idea of cycling over the colours.  And if we do that, I'd add a seventh colour (black?) so that there's a better chance of the numbers of lines being relatively prime.  :)

**Comment by jgosmann** [comment]
I don't see the advantage of a seventh color. Can you elaborate?

**Comment by tcstewar** [comment]
> I don't see the advantage of a seventh color. Can you elaborate?

It very slightly deals with the "they will repeat and might match up" problem.  So if I have 4 2D plots and 6 colours, then the first plot and the last plot will use the identical colour scheme.  The same thing will happen if I have 3 3D plots.  If we had 7 colours, then you'd need to have 8 2D plots (or 8 3D plots) before you found 2 with the identical colour scheme.

**Comment by jgosmann** [comment]
But you still get a repeated color with 4 2D plots and 7 colors in the first and last plot which is still a problem if you overlay them.

**Comment by tcstewar** [comment]
> But you still get a repeated color with 4 2D plots and 7 colors in the first and last plot which is still a problem if you overlay them.

Yes, but at least the plots themselves will be visually distinguishable based on what 2 colours are in them up until you get 8 plots.  That's all I was meaning here.

**Comment by jgosmann** [comment]
As long as we don't have any legend, I would prefer to have the same color. Then I can get away by just remembering the order of the first 3 colours in most cases.

**Comment by tcstewar** [comment]
> As long as we don't have any legend, I would prefer to have the same color. Then I can get away by just remembering the order of the first 3 colours in most cases.

I definitely agree.  However we do this, if we're not going to have all graphs have a standard colour order,  we need a way to get at legend indicating the colours.  Perhaps something on mouseover?  I'm thinking something that appears beside the plot, so that it doesn't block the view of anything in that plot.....

**Comment by tcstewar** [comment]
After a bit of discussion in the dev meeting, the plan going forward is something like this:

- Adding a per-plot option of changing the colour theme helps for this, as is happening in #571 
- It'd also be helpful to add a per-plot option of changing the starting colour in the theme
- We probably don't want a full-fledged colour-picker for complete colour customization, just due to the code complexity that would add, for fairly little payoff (in most cases, the above two options would be enough)
- In the long run, merging plots together ( #850 ) seems like a pretty good solution to this as well.
